app/src/main/java/ui/ScheduleActivity.kt

```diff
@@ -554,11 +554,10 @@ class ScheduleActivity : DrawerActivity() {
                 textColor = color
                 onClick {
                     // Try to find a place that has the right name
-                    launch(Dispatchers.IO) {
-                        val places = placeDao.getPlaces()
-                        val place = places.firstOrNull { place ->
-                            course.location.contains(place.name, true)
-                        }
+                    launch(ioDispatcher) {
```

Comment:
I know you had said that you first launch within the `uiDispatcher`, which I'm not doing here. What are your recommendations on how to rework this code to respect that?

Reply:
Essentially, it never hurts to add `withContext` to your suspend functions so each action itself specifies whether it is ui or io. The one case where the called may specify context is if you are wrapping a non suspended function, as the method is used without coroutine scopes elsewhere. So here I'd make loadPlaces a suspend function with io context

Reply:
Ok got it, I'll give it a shot.